### PR TITLE
[NFI]: Refactor ElementwiseOpToLLVM.cpp

### PR DESCRIPTION
**Pull request by etiotto**
Make file closer to NVidia implementation

### CODE REVIEW

**Review by chengjunlu — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 1]
```diff
@@ -2232,97 +1812,38 @@ void populateElementwiseOpToLLVMPatterns(
...
   // ExpOpConversionApprox will try using ex2.approx if the input type is
   // FP32. For other input types, ExpOpConversionApprox will return failure and
   // ElementwiseOpConversion<math::ExpOp, math::ExpOp> defined below will call
   // a vendor specific math library for higher-precision calculation
   patterns.add<ExpOpConversionApprox>(typeConverter, axisInfoAnalysis, benefit);
-  patterns.add<MulhiUIOpConversion>(typeConverter, axisInfoAnalysis, targetInfo,
-                                    benefit);
+  patterns.add<MulhiUIOpConversion>(
+      typeConverter, axisInfoAnalysis, targetInfo,
+      benefit); // must keep (is different, we hass SPIRV calling convention to
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1845]: is the comment intentional to stay?
```suggestion
      benefit); // must keep (is different, we have SPIRV calling convention to
```

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 2]
```diff
@@ -1249,10 +1222,12 @@ struct FpToFpOpConversion
     auto F64TyID = TypeID::get<Float64Type>();
 
     if (srcTy.getTypeID() == dstTy.getTypeID()) {
-      if (srcTy.getTypeID() == F8E4M3TyID || dstTy.getTypeID() == F8E4M3TyID)
-        return {identity_func, 2};
-      else
-        return {identity_func, 4};
+      auto identityFn = [](Location loc, ConversionPatternRewriter &rewriter,
+                           const SmallVector<Value> &v) { return v; };
```
Comment (victor-eds) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1226]:
```suggestion
      constexpr auto identityFn = [](Location, ConversionPatternRewriter &,
                           const SmallVector<Value> &v) { return v; };
```

Avoid warnings

**Review by victor-eds — COMMENTED** [review]
LGTM but potential warning

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1845]: Removed.

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1226]: Thanks. Added `constexpr`

File: third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp [hunk 3]
```diff
@@ -1222,8 +1222,9 @@ struct FpToFpOpConversion
     auto F64TyID = TypeID::get<Float64Type>();
 
     if (srcTy.getTypeID() == dstTy.getTypeID()) {
-      auto identityFn = [](Location loc, ConversionPatternRewriter &rewriter,
-                           const SmallVector<Value> &v) { return v; };
+      constexpr auto identityFn = [](Location,
+                                     ConversionPatternRewriter &rewriter,
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/ElementwiseOpToLLVM.cpp:1226]: `rewriter` can be removed too.

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**